crates/rust-analyzer/src/lib.rs

```diff
@@ -132,8 +133,16 @@ fn completion_item_hash(item: &CompletionItem, is_ref_completion: bool) -> [u8;
         hasher.update(detail);
     }
     hash_completion_relevance(&mut hasher, &item.relevance);
-    if let Some((mutability, text_size)) = &item.ref_match {
-        hasher.update(mutability.as_keyword_for_ref());
+    if let Some((ref_mode, text_size)) = &item.ref_match {
+        match ref_mode {
+            CompletionItemRefMode::Reference(mutability) => match mutability {
+                Mutability::Shared => hasher.update("&"),
+                Mutability::Mut => hasher.update("&mut"),
+            },
+            CompletionItemRefMode::Dereference => {
+                hasher.update("*");
+            }
+        }
```

Comment:
Similar to above, we can move just return the strings in the match, then call `hasher.update` on that